src/transformers/trainer.py

```diff
@@ -3595,13 +3635,13 @@ def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor,
 
         return loss.detach() / self.args.gradient_accumulation_steps
 
-    def compute_loss(self, model, inputs, return_outputs=False):
+    def _compute_loss(self, model, inputs, return_outputs=False, num_items_in_batch=None):
         """
         How the loss is computed by Trainer. By default, all models return the loss in the first element.
 
         Subclass and override for custom behavior.
         """
-        if self.label_smoother is not None and "labels" in inputs:
+        if (self.label_smoother is not None or self.compute_loss is not None) and "labels" in inputs:
             labels = inputs.pop("labels")
```

Comment:
mmmm if people don't pass a loss, we won't use the model's default?

Reply:
We will, it stays in `inputs` and gets passed to the models `forward()`